src/witan/httpapi/components/activities.clj

```diff
@@ -314,12 +343,16 @@
 (defn get-metadatastore-id
   [payload]
   (or (get-in payload [::ms/id])
+      (get-in payload [:kixi.datastore.metadatastore.relaxed/id])
       (get-in payload [:original ::ms/id])
       (get-in payload [:original ::ms/payload :kixi.comms.command/payload ::ms/id])))
 
 (defmulti on-event
-  (fn [_ {:keys [kixi.comms.event/key
-                 kixi.comms.event/version]}] [key version]))
+  (fn [_ e]
+    [(or (:kixi.comms.event/key e)
+         (:kixi.event/type e))
+     (or (:kixi.comms.event/version e)
+         (:kixi.event/version e))]))
```

Comment:
it always makes me a bit sad when we do these or checks rather than switching on something more more specific, but I understand why we do it.